### ISSUE TITLE
CPU info: Not yet implemented yet on Apple M1 Pro 

### ISSUE THREAD

**Issue by PeterBocan** [thought-machine/please#2514]
On my 14" Macbook Pro with Please version 16.21.3 I am getting this slightly annoying (makes `plz build --shell //target` unusable, due to constant printing into the console). 

```
22:10:27.461   ERROR: Error getting CPU info: not implemented yet
22:10:27.960   ERROR: Error getting CPU info: not implemented yet
22:10:28.461   ERROR: Error getting CPU info: not implemented yet
22:10:28.960   ERROR: Error getting CPU info: not implemented yet
22:10:29.460   ERROR: Error getting CPU info: not implemented yet
22:10:29.960   ERROR: Error getting CPU info: not implemented yet
22:10:30.461   ERROR: Error getting CPU info: not implemented yet
```

**Comment by PeterBocan** [comment]
As I did a little investigation in #2515 the `cpu.Times(false)` calls `allCPUTimes()` which, depending on whether the CGO is enabled or not, has two different implementations:

https://github.com/shirou/gopsutil/blob/2f8da0a39487ceddf44cebe53a1b563b0b7173cc/cpu/cpu_darwin_nocgo.go
https://github.com/shirou/gopsutil/blob/2f8da0a39487ceddf44cebe53a1b563b0b7173cc/cpu/cpu_darwin_cgo.go

It seems like something went wrong with the cross-compilation and CGO not being enabled for Apple plaftorm.

**Comment by alaroldai** [comment]
I see that #2515 has been merged, but I'm still seeing this on my M1 Mac Mini with Please 16.23.1:

```
13:56:37.567   ERROR: Error getting CPU info: not implemented yet
```

Are there any plans to address this? It also makes `plz watch` unusable.

**Comment by namtzigla** [comment]
I'm not sure wh but all the binaries for darwin_amd64 downloaded from github main repo, including 16.23.1, print out this error `ERROR: Error getting CPU info: not implemented yet` but if I'm compiling please (v16.231.) on my local macos m1 computer the error disappear.

I'm using a macos 13.0 with xcode  14.1 (14B47b).

**Comment by namtzigla** [comment]
new symptom, version 16.19.0 is the last known version where this does not happen.

**Comment by peterebden** [comment]
I can see it's finding `cpu_darwin_nocgo.go`. I suspect this is because of cross-compilation (we don't have a `darwin_arm64` worker on CircleCI) although not certain of all the logic there - @Tatskaari might know more

**Comment by peterebden** [comment]
It should also be possible to disable the display with the `display.systemstats` config option, which would at least stop the irritating messages in the short term.

**Comment by Tatskaari** [comment]
I made an attempt to fix this here: #1978

I think this worked, and indeed there are a number of releases where we can get CPU info. I'm not sure why this is occurring again. Perhaps an upstream update?

**Comment by Tatskaari** [comment]
Just checked and it seems to be fixed in v16.24.0.
